File: Iter.Api/Iter.Core/Models/Destination.cs
namespace Iter.Core.Models
{
    public class Destination
    {
        public Guid Id { get; set; }

        public string City { get; set; }

        public string Country { get; set; }

        public Guid AccommodationId { get; set; }

        public Accommodation Accommodation { get; set; }

        public Guid ArrangementId { get; set; }

        public Arrangement Arrangement{ get; set; }

        public DateTime ArrivalDate { get; set; }

        public DateTime DepartureDate { get; set; }

        public bool IsOneDayTrip { get; set; }


        public bool IsDeleted { get; set; }

        public DateTime DateCreated { get; set; }

        public DateTime DateModified { get; set; }
    }
}

File: Iter.Api/ui/lib/models/destination.dart
import 'package:ui/models/accomodation.dart';
import 'package:json_annotation/json_annotation.dart';

part 'destination.g.dart';

@JsonSerializable()
class Destination {
  String? id;
  String? city;
  int? cityId;
  int? countryId;
  String? country;
  DateTime? arrivalDate;
  DateTime? departureDate;
  bool? isOneDayTrip;
  Accomodation? accommodation;

  Destination(
      {
      this.id,
      this.city,
      this.country,
      this.cityId,
      this.countryId,
      this.arrivalDate,
      this.departureDate,
      this.isOneDayTrip,
      this.accommodation});

  factory Destination.fromJson(Map<String, dynamic> json) =>
      _$DestinationFromJson(json);
  Map<String, dynamic> toJson() => _$DestinationToJson(this);
}
